include/SFML/Window/Cursor.hpp

```diff
@@ -0,0 +1,214 @@
+    /// the current cursor is left unchanged and the function will
+    /// return false.
+    ///
+    /// In addition to specifying the pixel data, you can also
+    /// specify the location of the hotspot of the cursor. The
+    /// hotspot is the pixel coordinate within the cursor image
+    /// which will be located exactly where the mouse pointer
+    /// position is. Any mouse actions that are performed will
+    /// return the window/screen location of the hotspot.
+    ///
+    /// \param pixels   Array of pixels of the image
+    /// \param size     Width and height of the image
+    /// \param hotspot  (x,y) location of the hotspot
+    /// \return true if the cursor was successfully loaded;
+    ///         false otherwise
+    ///
+    ////////////////////////////////////////////////////////////
+    bool loadFromPixels(const Uint8* pixels, Vector2u size, Vector2u hotspot);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Create a native system cursor
+    ///
+    /// Refer to the list of cursor available on each system
+    /// (see sf::Cursor::Type) to know whether a given cursor is
+    /// expected to load successfully or is not supported by
+    /// the operating system.
+    ///
+    /// \param type Native system cursor type
+    /// \return true if and only if the corresponding cursor is
+    ///         natively supported by the operating system;
+    ///         false otherwise
+    ///
+    ////////////////////////////////////////////////////////////
+    bool loadFromSystem(Type type);
+
+private:
+
+    friend class Window;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get access to the underlying implementation
+    ///
+    /// This is primarily designed for sf::Window::setMouseCursor,
+    /// hence the friendship.
+    ///
+    /// \return a reference to the OS-specific implementation
+    ///
+    ////////////////////////////////////////////////////////////
+    const priv::CursorImpl& getImpl() const;
+
+private:
+
+    ////////////////////////////////////////////////////////////
+    // Member data
+    ////////////////////////////////////////////////////////////
+    priv::CursorImpl* m_impl; ///< Platform-specific implementation of the cursor
+};
+
+} // namespace sf
+
+
+#endif // SFML_CURSOR_HPP
+
+
+////////////////////////////////////////////////////////////
+/// \class sf::Cursor
+/// \ingroup window
+///
+/// \warning Features related to Cursor are not supported on
+///          iOS and Android.
+///
+/// This class abstracts the operating system resources
+/// associated with either a native system cursor or a custom
+/// cursor.
+///
+/// After loading the cursor the graphical appearance
+/// with either loadFromPixels() or loadFromSystem(), the
+/// cursor can be changed with sf::Window::setMouseCursor().
+///
+/// \todo Does Windows or Linux requires the Cursor instance
+/// to outlive it's usage?
```

Comment:
Also, I was not sure whether there was such limitations on Linux/Windows... For once, there's none on OS X.